presto-main/src/test/java/io/prestosql/sql/gen/TestExpressionCompiler.java

```diff
@@ -621,6 +621,35 @@ public void testBinaryOperatorsString()
         Futures.allAsList(futures).get();
     }
 
+    @Test
+    public void testNestedColumnFilter()
```

Comment:
Make sure this test fails without the changes to `SqlTypeBytecodeExpression.java`

Reply:
Sure @findepi. Setting `FunctionAssertions.TEST_ROW_NUMBER_OF_FIELDS` to a large number (eg. `4000`) reproes the issue. I didn't include it because `FunctionAssertions` is used in lot of tests, not sure of the implications of the having such large schema on test runtimes. Let me see if I can separate out the wide row into a separate test data variable.

Reply:
@findepi Updated commit to repro the issue in test case. I tried to find a better way to simplify the test using `CAST( ROW(1.0, 2.0, .... n) AS ROW(c1 DOUBLE, c2 DOUBLE,... cn DOUBLE))`. This reproes the issue, but hits another limit (generated method for `CAST` is too big). I ran the `TestExpressionComplier` and `TestRowOperators` with and without the large row data and I only see increase of `ms` in test runtime. Hope this is ok. Let me know if there is a simpler way to test this.